terminal/terminal.md

```diff
@@ -0,0 +1,417 @@
+
+We have been exploring how to move across directories, list files and see the metadata of those, but now, let's go one step ahead and create a directory with the **mkdir** command
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir mydirectory
+danielvilla@Daniels-MacBook-Air:~$ 
+```
+Normally when a command is successful it won't print anything, if we look now the mydirectory directory is created and ready to have files
+
+It's very common that we want to create directories contained inside directories (that's how everything works) but mkdir alone won't let us do that
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir great_ideas/best_ideas/dontlookhere
+mkdir: great_ideas/best_deas: No such file or directory
+```
+We need to add the **p** flag to correct that
+
+```
+danielvilla@Daniels-MacBook-Air:~$ mkdir -p great_ideas/best_ideas/dontlookhere
+danielvilla@Daniels-MacBook-Air:~$ 
+```
+That will save us a lot of time of doing `mkdir great_ideas` then `cd great_ideas` then `mkdir best_ideas`... well you get the point
+
+NOW! we have our directory ready to have files and great ideas, but how do we create a simple file? 
+
+There are plenty of ways to create files on unix system but the common ones are
+
+```
+danielvilla@Daniels-MacBook-Air:~$ touch idea1.txt
+```
+**touch** is an awesome command, you don't have to worry if the file exists and if it is going to get overwritten, if there's a idea1.txt file only the created date wil be affected to the current date, the data will stay intact
+
+
+**vim** is a text editor that will let you create files, the comprehension and usage of vim is WAY beyond this course, but I want to let you know people who know how to use vim are VERY efficient in matters of text insertion and deletion, vim alows you to do very delicated and elegant modifications to a file. For now lets just create a file
+
+```
+danielvilla@Daniels-MacBook-Air:~$ vim idea2.txt
+```
+vim will create idea2.txt, to enter in insert mode text press **i** 
+
+To exit press **esc** then **:** and then **q** follow by **enter** key
+
+```
+danielvilla@Daniels-MacBook-Air:~$ > idea3.txt
+```
+redirecting the output of nothing, it will create an empty text file
+
+To add text to a file you can simple vim into it and add the text in insert mode, or you can append text, echoing the text and adding it to the file
+
+```
+danielvilla@Daniels-MacBook-Air:~$ echo "create a script to automate installations" >> idea1.txt
+```
+to display what a file has let's use the cat command
+```
+danielvilla@Daniels-MacBook-Air:~$ cat idea1.txt 
+create a script to automate installations
+```
+we can also use the **less** and **more** commands, try it!
+
+#### Removing files and directories 
+
+Now that we have created a file and a directory let's learn how to remove it (we will be doing that a lot)
+```
+danielvilla@Daniels-MacBook-Air:~$ rm bad_dea.txt
+```
+as simple as that, but BE AWARE, there's trash can on the terminal, **removing something IS FOREVER**
```

Comment:
as simple as that, but BE AWARE, there's `is not` trash can ~on~ `in` the terminal, **removing something IS FOREVER**